### PR TITLE
fix(watcher): maxBlockAdvance and confidenceInterval 

### PR DESCRIPTION
**Pull request by TheAncientGoat**
This adds two flags to the watcher, to prevent over-subscribing to event logs via a configurable limit, and to prevent fetching the latest blocks to avoid re-shuffling and missing events in the latest blocks.

- [x] Add configuration values via constructor
- [x] Add configuration values via config
- [ ] Test on devnet

### CODE REVIEW

**Review by jazg — CHANGES_REQUESTED** [review]
Looks good! Just requested some small changes.

**Comment by jazg** [comment]
@TheAncientGoat Can you please update the changelog as well?

File: watcher/watcher.go [hunk 1]
```diff
@@ -104,34 +174,41 @@ func (watcher Watcher) watchLogShiftOuts(parent context.Context) {
...
-	iter, err := watcher.ethBindings.FilterLogBurn(
-		&bind.FilterOpts{
-			Context: ctx,
-			Start:   last + 1, // Add one to avoid duplication.
-			End:     &cur,
-		},
-		nil,
-		nil,
-	)
+	// Only advance by a set number of blocks at a time to prevent over-subscription
```
[the remark this answers is not in the document]
Comment (TheAncientGoat) [reply watcher/watcher.go:177]: Have asked on the go-eth discord; but couldn't find it documented anywhere. We're currently doing 15(tps) * 15(s interval) -  1k is less than an order of magnitude increase, so it should be safe; but still allow us to catch up at a rate of (downtime (s)*15(tps)) / 1000

[the remark this answers is not in the document]
Comment (jazg) [reply watcher/watcher.go:179]: Sounds good

File: watcher/watcher_test.go [hunk 2]
```diff
@@ -2,26 +2,104 @@ package watcher_test
...
+	v0 "github.com/renproject/lightnode/compat/v0"
 	. "github.com/renproject/lightnode/watcher"
 
-	"github.com/alicebob/miniredis"
+	"github.com/alicebob/miniredis/v2"
 	"github.com/go-redis/redis/v7"
 	"github.com/renproject/darknode/jsonrpc/jsonrpcresolver"
 	"github.com/renproject/darknode/tx"
 	"github.com/renproject/darknode/txengine/txenginebindings"
+	"github.com/renproject/darknode/txengine/txenginebindings/ethereumbindings"
```
[the remark this answers is not in the document]
Comment (jazg) [reply watcher/watcher_test.go]: Not particularly, usually just have the same types of imports grouped together (e.g. Go standard imports -> GitHub imports -> `.`/`_` imports)

[the remark this answers is not in the document]
Comment (jazg) [reply watcher/watcher_test.go]: Also for future reference, function/object comments should lead with the name, e.g. `FetchBurnLogs` might start with `// FetchBurnLogs for...`

**Review by jazg — APPROVED** [review]

**Pull request merged**